controller/src/test/java/io/pravega/controller/server/rpc/auth/ControllerGrpcAuthFocusedTest.java

```diff
@@ -627,37 +625,16 @@ public void onNext(T value) {
 
         @Override
         public void onError(Throwable t) {
-            synchronized (this) {
-                error = t;
-                completed.set(true);
-                this.notifyAll();
-            }
+            future.completeExceptionally(t);
         }
 
         @Override
         public void onCompleted() {
-            synchronized (this) {
-                completed.set(true);
-                this.notifyAll();
-            }
+            future.complete(result);
         }
 
-        @SneakyThrows
-        public T get() {
-            synchronized (this) {
-                while (!completed.get()) {
-                    try {
-                        this.wait();
-                    } catch (InterruptedException e) {
-                        return null;
-                    }
-                }
-            }
-            if (error != null) {
-                throw error;
-            } else {
-                return result;
-            }
+        public T get() throws ExecutionException, InterruptedException {
+            return future.get();
```

Comment:
Would it make sense to use `join` instead of `get` so that we don't have to declare that it throws?

Reply:
Yes, that's a good idea. Using `join()` instead of `get()` would make the code simpler here. I've modified it as suggested.